cranelift/codegen/src/ir/jumptable.rs

```diff
@@ -114,24 +128,34 @@ mod tests {
         assert_eq!(jt.as_slice().get(0), None);
         assert_eq!(jt.as_slice().get(10), None);
 
-        assert_eq!(jt.to_string(), "block0, []");
+        assert_eq!(jt.display(&pool).to_string(), "block0, []");
 
         assert_eq!(jt.all_branches(), [def]);
         assert_eq!(jt.as_slice(), []);
     }
 
     #[test]
     fn insert() {
-        let def = Block::new(0);
+        let mut pool = ValueListPool::default();
+
+        let e0 = Block::new(0);
         let e1 = Block::new(1);
         let e2 = Block::new(2);
 
-        let jt = JumpTableData::new(def, &[e1, e2, e1]);
+        let def = BlockCall::new(e0, &[], &mut pool);
+        let b1 = BlockCall::new(e1, &[], &mut pool);
+        let b2 = BlockCall::new(e2, &[], &mut pool);
+        let b3 = BlockCall::new(e1, &[], &mut pool);
+
+        let jt = JumpTableData::new(def, &[b1, b2, b3]);
 
         assert_eq!(jt.default_block(), def);
-        assert_eq!(jt.to_string(), "block0, [block1, block2, block1]");
+        assert_eq!(
+            jt.display(&pool).to_string(),
+            "block0, [block1, block2, block1]"
+        );
 
-        assert_eq!(jt.all_branches(), [def, e1, e2, e1]);
-        assert_eq!(jt.as_slice(), [e1, e2, e1]);
+        assert_eq!(jt.all_branches(), [def, b1, b2, b3]);
+        assert_eq!(jt.as_slice(), [b1, b2, b3]);
```

Comment:
Maybe also add a test using block arguments?

Reply:
I've added both runtests and x64 compile tests for uses of `br_table` with block arguments. My feeling was that those would be sufficient as this PR stops splitting critical edges in ssa construction, was there an additional case you were thinking of here?

Reply:
I'll add different arguments to the two uses of `e1`, and assert that they're the same.